src/Foundatio/Jobs/IJob.cs

```diff
@@ -40,32 +40,29 @@ public static class JobExtensions {
                         else if (iterations % 1000 == 0) // allow for cancellation token to get set
                             await SystemClock.SleepAsync(1).AnyContext();
                     } catch (TaskCanceledException) { }
-
-                    if (continuationCallback == null)
+                    
+                    if (continuationCallback == null || cancellationToken.IsCancellationRequested)
                         continue;
-
+                    
                     try {
                         if (!await continuationCallback().AnyContext())
                             break;
                     } catch (Exception ex) {
                         logger.Error(ex, "Error in continuation callback: {0}", ex.Message);
                     }
                 }
-
-                logger.Info("Stopping continuous job type \"{0}\" on machine \"{1}\"...", jobName, Environment.MachineName);
-
+                
                 if (cancellationToken.IsCancellationRequested)
                     logger.Trace("Job cancellation requested.");
 
-                await SystemClock.SleepAsync(1).AnyContext(); // allow events to process
+                logger.Info("Stopping continuous job type \"{0}\" on machine \"{1}\"...", jobName, Environment.MachineName);
```

Comment:
It doesn't make any sense to sleep after a continuous run is completed. If anything this should be inside of the while loop